src/commands/launchDebugProxy.ts

```diff
@@ -0,0 +1,50 @@
+import * as vscode from 'vscode';
+import { logger, config } from '../extension';
+import type { CloudAppNode } from '../CloudDataProvider';
+import { launchDebugProxy } from '../debugProxy';
+import { getDbInstance, isUnauthorized } from '../dbosCloudApi';
+import { validateCredentials } from '../validateCredentials';
+import { DbosDebugConfig } from '../Configuration';
+
+function isDebugConfig(node: CloudAppNode | DbosDebugConfig): node is DbosDebugConfig {
+  return (node as DbosDebugConfig).host !== undefined;
+}
+
+export function getLaunchDebugProxyCommand(storageUri: vscode.Uri) {
+
+  return async function (arg?: CloudAppNode | DbosDebugConfig): Promise<boolean> {
+    logger.debug("launchDebugProxy", { app: arg ?? null });
+    if (!arg) { return false; }
+
+    let debugConfig: DbosDebugConfig;
+    if (isDebugConfig(arg)) {
+      debugConfig = arg;
+    } else {
+      const credentials = await config.getStoredCloudCredentials(arg.domain);
+      if (!validateCredentials(credentials)) { return false; }
```

Comment:
As mentioned above, I think in general, we need better error message/prompt from the extension. Sometimes when I tried to demo time travel the proxy would fail to launch without displaying any errors.
If we `return false` here, does it fail to start silently?

Reply:
validateCredentials launches login flow if the credentials are undefined or expired.